### PR TITLE
HBASE-22803 Modify config value range to enable turning off of the hbck chore

### PR DESCRIPTION
**Pull request by jatsakthi**

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java [hunk 1]
```diff
@@ -98,29 +98,49 @@
   private volatile long checkingStartTimestamp = 0;
   private volatile long checkingEndTimestamp = 0;
 
+  private boolean stopped = false;
```
Comment (infraio) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java]: stopped ==> disabled?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java [hunk 2]
```diff
@@ -98,29 +98,49 @@
...
+
   public HbckChore(MasterServices master) {
     super("HbckChore-", master,
         master.getConfiguration().getInt(HBCK_CHORE_INTERVAL, DEFAULT_HBCK_CHORE_INTERVAL));
     this.master = master;
+    int interval =
+        master.getConfiguration().getInt(HBCK_CHORE_INTERVAL, DEFAULT_HBCK_CHORE_INTERVAL);
+    if (interval <= 0) {
+      LOG.warn(HBCK_CHORE_INTERVAL + "is <=0 hence disabling hbck chore");
```
Comment (infraio) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java]: Need a whitespace before "is".

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 75 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 449 | master passed |
| +1 | compile | 70 | master passed |
| +1 | checkstyle | 97 | master passed |
| +1 | shadedjars | 345 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 44 | master passed |
| 0 | spotbugs | 312 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 309 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 396 | the patch passed |
| +1 | compile | 67 | the patch passed |
| +1 | javac | 67 | the patch passed |
| +1 | checkstyle | 94 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 340 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1193 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 40 | the patch passed |
| +1 | findbugs | 307 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 18277 | hbase-server in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 22597 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAsyncTableGetMultiThreaded |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestCloneSnapshotFromClientNormal |
|   | hadoop.hbase.client.TestFromClientSide |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/466 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux aafcab0cb35c 4.4.0-154-generic #181-Ubuntu SMP Tue Jun 25 05:29:03 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-466/out/precommit/personality/provided.sh |
| git revision | master / 547cec4078 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/1/testReport/ |
| Max. process+thread count | 4622 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 66 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 352 | master passed |
| +1 | compile | 57 | master passed |
| +1 | checkstyle | 90 | master passed |
| +1 | shadedjars | 295 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 37 | master passed |
| 0 | spotbugs | 266 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 264 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 327 | the patch passed |
| +1 | compile | 57 | the patch passed |
| +1 | javac | 57 | the patch passed |
| +1 | checkstyle | 87 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 297 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1020 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 34 | the patch passed |
| +1 | findbugs | 273 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 13401 | hbase-server in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 17099 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/466 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux b690cd16ef2b 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-466/out/precommit/personality/provided.sh |
| git revision | master / 9250977681 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/3/testReport/ |
| Max. process+thread count | 4892 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 140 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 363 | master passed |
| +1 | compile | 57 | master passed |
| +1 | checkstyle | 83 | master passed |
| +1 | shadedjars | 313 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 36 | master passed |
| 0 | spotbugs | 262 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 259 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 393 | the patch passed |
| +1 | compile | 58 | the patch passed |
| +1 | javac | 58 | the patch passed |
| +1 | checkstyle | 80 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 321 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1112 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 35 | the patch passed |
| +1 | findbugs | 289 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 18344 | hbase-server in the patch failed. |
| +1 | asflicense | 37 | The patch does not generate ASF License warnings. |
| | | 22342 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestReplicationSmallTestsSync |
|   | hadoop.hbase.tool.TestBulkLoadHFiles |
|   | hadoop.hbase.replication.TestReplicationDisableInactivePeer |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.tool.TestSecureBulkLoadHFiles |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/466 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 46369aa56fb5 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-466/out/precommit/personality/provided.sh |
| git revision | master / 9250977681 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/2/testReport/ |
| Max. process+thread count | 4936 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by jatsakthi** [comment]
Thanks for your suggestions @infraio , have update the changes.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 75 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 363 | master passed |
| +1 | compile | 60 | master passed |
| +1 | checkstyle | 91 | master passed |
| +1 | shadedjars | 305 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 40 | master passed |
| 0 | spotbugs | 273 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 270 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 326 | the patch passed |
| +1 | compile | 56 | the patch passed |
| +1 | javac | 56 | the patch passed |
| +1 | checkstyle | 90 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 295 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1022 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 35 | the patch passed |
| +1 | findbugs | 266 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 13701 | hbase-server in the patch failed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 17439 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/466 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f7a93e6c26e0 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-466/out/precommit/personality/provided.sh |
| git revision | master / 9250977681 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/4/testReport/ |
| Max. process+thread count | 4932 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-466/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by jatsakthi** [comment]
Thanks for the review @infraio

**Pull request merged**